openvdb/cmd/CMakeLists.txt

```diff
@@ -0,0 +1,296 @@
+# Mozilla Public License 2.0 ( http://www.mozilla.org/MPL/2.0/ )
+#
+# Redistributions of source code must retain the above copyright
+# and license notice and the following restrictions and disclaimer.
+#
+# *     Neither the name of DreamWorks Animation nor the names of
+# its contributors may be used to endorse or promote products derived
+# from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDERS' AND CONTRIBUTORS' AGGREGATE
+# LIABILITY FOR ALL CLAIMS REGARDLESS OF THEIR BASIS EXCEED US$250.00.
+#
+#[=======================================================================[
+
+  CMake Configuration for OpenVDB Binaries
+
+#]=======================================================================]
+
+PROJECT ( OpenVDBBinaries )
+CMAKE_MINIMUM_REQUIRED ( VERSION 3.3 )
+# Monitoring <PackageName>_ROOT variables
+IF ( POLICY CMP0074 )
+  CMAKE_POLICY ( SET CMP0074 NEW )
+ENDIF ()
+
+###### OpenVDB Binary Component Options
+
+OPTION ( OPENVDB_BUILD_VDB_LOD "Build vdb_lod" ON )
+OPTION ( OPENVDB_BUILD_VDB_PRINT "Build vdb_print" ON )
+OPTION ( OPENVDB_BUILD_VDB_RENDER "Build vdb_render" ON )
+OPTION ( OPENVDB_BUILD_VDB_VIEW "Build vdb_view" ON )
+
+#########################################################################
+
+MESSAGE ( STATUS "----------------------------------------------------" )
+MESSAGE ( STATUS "----------- Configuring OpenVDBBinaries ------------" )
+MESSAGE ( STATUS "----------------------------------------------------" )
+
+##########################################################################
+
+# Collect and configure lib dependencies
+
+IF ( NOT OPENVDB_BUILD_CORE )
+  FIND_PACKAGE ( OpenVDB REQUIRED )
+  SET ( OPENVDB_LIB OpenVDB::openvdb )
+ELSE ()
+  SET ( OPENVDB_LIB openvdb_shared )
+ENDIF ()
+
+FIND_PACKAGE ( IlmBase REQUIRED COMPONENTS Half Iex IlmThread )
+FIND_PACKAGE ( TBB REQUIRED COMPONENTS tbb )
+FIND_PACKAGE ( ZLIB REQUIRED )
+
+IF ( USE_LOG4CPLUS )
+  FIND_PACKAGE ( Log4cplus REQUIRED )
+ENDIF ()
+
+IF ( USE_BLOSC )
+  FIND_PACKAGE ( Blosc ${MINIMUM_BLOSC_VERSION} REQUIRED )
+ENDIF ()
+
+FIND_PACKAGE ( Boost ${MINIMUM_BOOST_VERSION} REQUIRED COMPONENTS iostreams system )
```

Comment:
Most of these dependencies are implicit in the fact that we link against libopenvdb - could these somehow just be specified in the global CMakeLists.txt so it doesn't need to be done in every subdir CMakeLists.txt? (Right now, I think every single library / application always links against the OpenVDB core library)

Reply:
Yeah I think this can be done - They should be set as interface libraries on openvdb_shared in openvdb/CMakeLists.txt (and ideally in FindOpenVDB too but I think that's going to be more tricky). I'll look to moving shared deps where possible.

Reply:
This is now done. The FindOpenVDB cmake module should now support both cases of building vdb components without building the core lib AND being used in external packages